packages/core/src/toolkit-provider/ToolkitProvider.tsx

```diff
@@ -137,9 +154,35 @@ export function ToolkitProvider({
     children,
     themes.map((theme) => theme.name),
     density,
+    applyClassesToBody,
     applyClassesToChild
   );
 
+  useIsomorphicLayoutEffect(() => {
+    if (applyClassesToBody) {
+      if (isRoot) {
+        // add the styles we want to apply
+        document.body.classList.add(
+          ...themes.map((theme) => `uitk-${theme.name}`),
+          `uitk-density-${density}`
+        );
+      } else {
+        console.warn(
+          "\nToolkitProvider can only apply CSS classes to the body if it is the root level ToolkitProvider."
+        );
+      }
+    }
+    return () => {
+      if (applyClassesToBody) {
+        // When unmounting/remounting, remove the applied styles from the body
+        document.body.classList.remove(
+          ...themes.map((theme) => `uitk-${theme.name}`),
```

Comment:
This map seems to happen in a few places e.g. also in `createThemedChildren`. Maybe we can just do it once and use the result in multiple places?